### PR TITLE
Add WindowProc delegation to Windows template

### LINKED ISSUE #53168: Support WindowProc delegation in plugin API
There are cases where plugin functionality requires changing the behavior of the WindowProc of the top-level HWND and/or the Flutter HWND. Examples:
- https://github.com/google/flutter-desktop-embedding/issues/483
- https://github.com/google/flutter-desktop-embedding/issues/636

We should allow plugins to register as delegates of the top-level WindowProc (if the app chooses to provide it; for add-to-app this would not always make sense, so we should be clear that it is not required to work) and the Flutter WindowProc (if we keep an HWND-based approach at that level).

### PR DESCRIPTION
**Pull request by stuartmorgan-g**
## Description

This wires up the new WindowProc delegation system that allows plugins
to handle top-level window messages (e.g., to control resize behavior).

## Related Issues

Fixes #53168

## Tests

I added the following tests: None (Windows e2e tests aren't available yet)

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by stuartmorgan** [comment]
This will need to wait until there's been an engine roll to land, since it uses new API.

**Review by jonahwilliams — APPROVED** [review]
RSLGTM

**Pull request merged**
